src/huntsman/pocs/camera/pyro/client.py

```diff
@@ -261,41 +277,6 @@ def _readout(self, **kwargs):
         """Dummy method on the client required to overwrite @abstractmethod"""
         pass
 
-    def _run_timeout(self, event_name, event_type, blocking, max_wait):
-        if blocking:
-            event = getattr(self, f"_{event_name}_event")
-            timer = CountdownTimer(duration=max_wait)
-            while not timer.expired():
-                if not event.is_set():
-                    return
-            self._timeout_response(event_name, event_type, max_wait, blocking)
-
-        else:
-            # If the remote operation fails after starting in such a way that the event doesn't
-            # get set then calling code could wait forever. Have a local timeout thread
-            # to be safe.
-            timeout_thread = Timer(interval=max_wait, function=self._timeout_response,
-                                   args=(event_name, event_type, max_wait, blocking))
-            timeout_thread.start()
```

Comment:
Is this not needed any more?

Reply:
@wtgee  I'm not sure why it would be. The `_focus_event` is a `RemoteEvent` object, so its timeout is  implicitly handled on by the camera service. The `_readout_thread` is not monitored in the `AbstractCamera`

Otherwise, the class should probably try and mimic the behaviour of `AbstractCamera` as much as possible, which doesn't spawn a new thread to monitor its `readout_thread` in `take_exposure`.

I guess the only reason we would keep something like this is to raise an error/warning on the client side if there is a timeout on the camera service. But this would be raised inside a different thread anyway.

Thoughts?